source/AI.cpp

```diff
@@ -2714,9 +2714,44 @@ void AI::DoSurveillance(Ship &ship, Command &command, shared_ptr<Ship> &target)
 		unsigned total = targetShips.size() + targetPlanets.size() + targetSystems.size();
 		if(!total)
 		{
-			// If there is nothing for this ship to scan, have it hold still
-			// instead of drifting away from the system center.
-			Stop(ship, command);
+			// If there is nothing for this ship to scan, have it patrol the entire system
+			// instead of drifting or stopping.
+			// Also allows the ship to land.
+			double radius = 1000. * 1000. * 1.1;
+			for(const StellarObject &object : ship.GetSystem()->Objects())
+				radius = max(radius, object.Position().LengthSquared() * 1.1);
```

Comment:
this doesnt feel necessary to me, seems unlikely they'd be further
I think we should either just base it on the objects, or on the arrival rate, or on the system limit gamerule, but we need to make a choice